cwltool/job.py

```diff
@@ -463,6 +466,25 @@ def run(self, pull_image=True, rm_container=True,
         # directory." but spec might change to designated temp directory.
         # runtime.append("--env=HOME=/tmp")
         runtime.append(u"--env=HOME=%s" % self.builder.outdir)
+        
+        # add parameters to docker to write a container ID file
+        if record_container_id:
+            if cidfile_dir:
+                if not os.path.isdir(cidfile_dir):
+                    _logger.error("--cidfile-dir %s error:\n%s", cidfile_dir,
+                                  cidfile_dir + " is not a directory or directory doesn't exist, please check it first")
+                    exit(-1)
+                if not os.path.exists(cidfile_dir):
+                    _logger.error("--cidfile-dir %s error:\n%s", cidfile_dir,
+                                  "directory doesn't exist, please create it first")
+                    exit(-1)
+            else:
+                cidfile_dir = "/tmp/"
+            cidfile_name = datetime.datetime.now().strftime("%Y%m%d%H%M%S-%f")+".cid"
+            if cidfile_prefix != "":
+                cidfile_name = str(cidfile_prefix + "-" + cidfile_name)
+            cidfile_path = cidfile_dir + cidfile_name
+            runtime.append(u"--cidfile=%s" % cidfile_path)
```

Comment:
Is there an equivalent for Singularity?

Reply:
Sorry but I don't think I understand this. Can you explain it a little bit.
I have finished other suggested changes

Reply:
http://singularity.lbl.gov/

If this option only works for the Docker runtime then we should say so. If we can add the equivalent option for Singularity then that would be nice.

Reply:
As far as I know, it comes with docker. I didn't know singularity before, but by reading its documents, I believe it's possible to add the equivalent option.